File: src/python/maps/map39.py
from browser import window
import ika

def AutoExec(engineRef):
    engineRef.mapThings.append(window.Snow.new(engineRef, 600, [.4, 1], [192,192,255]))
    engineRef.background = engineRef.getImage('gfx/mountains.png')

def to38(engineRef):
    offset_from = 4 * 16  # first vertical pos possible
    offset_to = 6 * 16  # first vertical pos possible
    x = engineRef.player.sprite.x - offset_from + offset_to
    yield from engineRef.mapSwitchTask('map38.ika-map', (x, 28 * 16))

def to40(engineRef):
    offset_from = 5 * 16  # first vertical pos possible
    offset_to = 8 * 16  # first vertical pos possible
    x = engineRef.player.sprite.x - offset_from + offset_to
    yield from engineRef.mapSwitchTask('map40.ika-map', (x, 1 * 16))

def to41(engineRef):
    offset_from = 34 * 16  # first vertical pos possible
    offset_to = 8 * 16  # first vertical pos possible
    y = engineRef.player.sprite.y - offset_from + offset_to
    yield from engineRef.mapSwitchTask('map41.ika-map', (1 * 16, y))

File: src/winter/maps/map39.ts
import { MapScript } from "./mapscript.js"
import { PyEngine, Snow } from "./../winter.js"

export default new MapScript(autoexec, { to38, to40, to41 })

export function autoexec(engineRef: PyEngine) {
    const engine = engineRef.getEngine()
    engineRef.addMapThing(new Snow(engineRef, 600, [.4, 1], [192,192,255]))
    engineRef.pySetBackground(engine.getImage('gfx/mountains.png'))
}

export function *to38(engineRef: PyEngine) {
    const offset_from = 4 * 16  // first vertical pos possible
    const offset_to = 6 * 16  // first vertical pos possible
    const p = engineRef.getPlayerEntity()
    const x = p.sprite.x - offset_from + offset_to
    yield* engineRef.mapSwitchTask('map38.ika-map', [x, 28 * 16])
}

export function *to40(engineRef: PyEngine) {
    const offset_from = 5 * 16  // first vertical pos possible
    const offset_to = 8 * 16  // first vertical pos possible
    const p = engineRef.getPlayerEntity()
    const x = p.sprite.x - offset_from + offset_to
    yield* engineRef.mapSwitchTask('map40.ika-map', [x, 1 * 16])
}

export function *to41(engineRef: PyEngine) {
    const offset_from = 34 * 16  // first vertical pos possible
    const offset_to = 8 * 16  // first vertical pos possible
    const p = engineRef.getPlayerEntity()
    const y = p.sprite.y - offset_from + offset_to
    yield* engineRef.mapSwitchTask('map41.ika-map', [1 * 16, y])
}
